#include <catch2/catch_test_macros.hpp>

#include "test_common.hpp"

#include <net/ServerSocket.hpp>
//#include <server/Server.hpp>

TEST_CASE( "ServerSocket test" ) {

//    const int port = GET_RANDOM_PORT;
//    hub::Server server( port );
//    server.setMaxClients( 0 );
//    server.asyncRun();

//    try {
//        hub::Server server2( port );
//        CHECK( false );
//    }
//    catch ( hub::net::Socket::exception& ex ) {
//        std::cout << "[test] catch exception : " << ex.what() << std::endl;
//        CHECK( true );
//    }
//    catch ( std::exception& ex ) {
//        std::cout << "[test] catch exception : " << ex.what() << std::endl;
//    }

//    hub::net::ServerSocket serverSocket;
}
